unlock-app/src/hooks/useLocks.js

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from 'react'
+import { Web3Service } from '@unlock-protocol/unlock-js'
+import { UNLIMITED_KEYS_COUNT } from '../constants'
+
+import GraphService from '../services/graphService'
+import configure from '../config'
+
+const config = configure()
+
+/**
+ * A hook which yields locks
+ */
+/**
+ * This hook reverse resolves any Ethereum address using the Ethereum Name Service
+ * @param {*} address
+ */
+export const useLocks = owner => {
+  // Let's retrieve the locks!
+  const [loading, setLoading] = useState(true)
+  const [locks, setLocks] = useState([])
+
+  const {
+    subgraphURI,
+    readOnlyProvider,
+    unlockAddress,
+    blockTime,
+    requiredConfirmations,
+    requiredNetworkId,
+  } = config
+  const graphService = new GraphService(subgraphURI)
+
+  const web3Service = new Web3Service({
```

Comment:
Not sure if it will matter in this case, but we might conceivably instantiate multiple `Web3Service`s if this hook (or others like it) gets wider use. If they do we might investigate making sure that Web3Service is a singleton that each hook can get a reference to.

Reply:
Good point. I think no state is shared in Web3Service so we're probably good.